configuration/config-txt/memory.md

```diff
@@ -2,31 +2,36 @@
 
-`gpu_mem` refers to memory that is addressable from the GPU, which includes the VPU, HVS, legacy codecs (e.g. H264), and camera, and on devices before the Raspberry Pi 4, the 3D system. The Raspberry Pi 4 3D system has it's own Memory Mangement Unit (MMU) so textures and other GL resources are not allocated from the `gpu_mem` but Linux system memory instead. This means that `gpu_mem` can be set to a lower value, so even if you are using the H264 and camera then 128MB will probably be enough. On earlier models without the 3D MMU, you may need up to 256 or 512 in some more unusual cases. 
+The recommended maximum values are as follows:
 
-For performance reasons, you should set `gpu_mem` as low as possible to give the Linux system as much memory as possible. However, setting `gpu_mem` to too low values may automatically disable certain firmware features, as there are some things the GPU cannot do if it has access to too little memory. So if a feature you are trying to use isn't working, try setting a larger GPU memory split.
+| total RAM | `gpu_mem` recommended maximum |
+|-----------|-------------------------------|
+| 256MB     | `128`                         |
+| 512MB     | `384`                         |
+| 1GB or greater | `512`                    |
 
-Values of `gpu_mem` over 512 are not recommended, will provide no performance improvements, and are untested.
+ It is possible to set `gpu_mem` to larger values, however this should be avoided since it can cause problems, such as preventing Linux from booting. There is no performance advantage to specifying values larger than is necessary. The minimum value is `16`, however this disables certain GPU features.
 
-Using `gpu_mem_256`, `gpu_mem_512`, and `gpu_mem_1024` allows you to swap the same SD card between 256MB, 512MB, and 1024MB Pis without having to edit `config.txt` each time:
+On the Raspberry Pi 4 the 3D component of the GPU has its own memory management unit (MMU), and does not use memory from the `gpu_mem` allocation. Instead memory is allocated dynamically within Linux. This may allow a smaller value to be specified for `gpu_mem` on the Pi 4, compared to previous models.
+
+You can also use `gpu_mem_256`, `gpu_mem_512`, and `gpu_mem_1024` to allow swapping the same SD card between Pis with different amounts of RAM without having to edit `config.txt` each time:
 
 ## gpu_mem_256
 
-The `gpu_mem_256` command sets the GPU memory in megabytes for the 256MB Raspberry Pi. (It is ignored if memory size is not 256MB). This overrides `gpu_mem`. The maximum value is `192`, and the default is not set.
+The `gpu_mem_256` command sets the GPU memory in megabytes for the 256MB Raspberry Pi. (It is ignored if memory size is not 256MB). This overrides `gpu_mem`.
```

Comment:
Minor nitpick, but should "the 256MB Raspberry Pi" be replaced with something like "Raspberry Pi devices with 256MB of memory" ? (and again below with 512MB)

Reply:
Good catch. I've used "Raspberry Pis" rather than "Raspberry Pi devices" as the extra word doesn't add anything, and I would rather keep it simple. It also fits in better with the intended chatty, informal style of official Pi material. (With the usual caveat that I'm not entirely convinced that is appropriate for technical docs).